net/socket/udp_socket_starboard.cc

```diff
@@ -159,6 +178,80 @@ int UDPSocketStarboard::GetLocalAddress(IPEndPoint* address) const {
   return OK;
 }
 
+int UDPSocketStarboard::StartWatchingSocketForReading() {
+  if (!base::CurrentIOThread::Get()->Watch(
+          socket_, true, base::MessagePumpIOStarboard::WATCH_READ,
+          &socket_watcher_, this)) {
+    PLOG(ERROR) << "WatchSocket failed on read";
+    Error result = MapLastSocketError(socket_);
+    if (result == ERR_IO_PENDING) {
+      // Watch(...) might call SbSocketWaiterAdd() which does not guarantee
+      // setting system error on failure, but we need to treat this as an
+      // error since watching the socket failed.
+      result = ERR_FAILED;
+    }
+    LogRead(result, NULL, NULL);
+    return result < 0 ? result : ERR_FAILED;
+  }
+  return OK;
+}
+
+int UDPSocketStarboard::ReadMultiplePackets(Socket::ReadPacketResults* results,
```

Comment:
This could use a bespoke unit test just for this function separately - but given the rest of the code doesn't seem to have it, i'd expect it will get tested at higher level. So leave as is.